com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -61,6 +61,53 @@ public static FailedCheck Error(string message)
             }
         }
 
+        /// <summary>
+        /// Checks that a model has the appropriate version.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters
+        /// </param>
+        /// <returns>A FailedCheck containing the error message if the version of the model does not mach, else null</returns>
+        public static FailedCheck CheckModelVersion(Model model)
+        {
+            var modelApiVersion = model.GetVersion();
+            if (modelApiVersion < (int)ModelApiVersion.MinSupportedVersion)
+            {
+                return FailedCheck.Error(
+                    "Model was trained with a older version of the trainer than is supported. " +
```

Comment:
Reworded these to be (IMHO) clearer and more actionable.

There's a lot of overlap between the first two messages, but I don't think it's worth trying to reduce it to a single case with string interpolation.